Project2015To2017.Migrate2017.Tool/CommandLogic.cs

```diff
@@ -1,89 +1,198 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
+using DotNet.Globbing;
 using Project2015To2017.Analysis;
 using Project2015To2017.Definition;
+using Project2015To2017.Reading;
 using Project2015To2017.Transforms;
 using Project2015To2017.Writing;
+using Serilog;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
 
 namespace Project2015To2017.Migrate2017.Tool
 {
 	public class CommandLogic
 	{
 		private readonly Microsoft.Extensions.Logging.ILogger genericLogger;
+		private ImmutableArray<(string path, string extension)> files;
+		private readonly ImmutableArray<string> extensions;
 
 		public CommandLogic()
 		{
 			this.genericLogger = new Serilog.Extensions.Logging.SerilogLoggerProvider().CreateLogger(nameof(Serilog));
+			extensions = ProjectConverter.ProjectFileMappings.Keys.Concat(new[] {".sln"}).ToImmutableArray();
 		}
 
-		public IReadOnlyCollection<Project> ParseProjects(
+		public void DoProcessableFileSearch(bool force = false)
```

Comment:
Why does this use a field, this could/should be side effect free, right? This could be private?

Reply:
Maybe I'm missing the point of what you're trying to do here. So you call this once to set a field, but I can never get hold of the value of that field. So how am I supposed to use this.

If it would return the value I could pass that in somewhere else and this would be the one place where you hit the filesystem.

Reply:
So I would pass that true to reevaluate. That means I am trying to handle the case where, while the tool is running, new solutions/projects were added under the current directory?

Reply:
One use can be that, yeah. But it might also be beneficial for possible future extension (or use of CLI tool as library, why the hell not?)